src/test/java/org/folio/rest/api/ItemShelvingOrderMigrationServiceApiTest.java

```diff
@@ -0,0 +1,90 @@
+package org.folio.rest.api;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
+import static org.folio.rest.api.StorageTestSuite.tenantOp;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.notNullValue;
+
+import io.vertx.core.json.JsonObject;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.folio.rest.jaxrs.model.Parameter;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.support.IndividualResource;
+import org.junit.Test;
+
+public class ItemShelvingOrderMigrationServiceApiTest extends MigrationTestBase {
+  @Test
+  public void shouldPopulateShelvingOrder() throws Exception {
+    var items = create201Items();
+    removeShelvingOrder(items);
+
+    var ta = new TenantAttributes()
+      .withModuleFrom("20.1.1")
+      .withParameters(List.of(
+        new Parameter().withKey("loadSample").withValue("false"),
+        new Parameter().withKey("loadReference").withValue("false")));
+
+    tenantOp(TENANT_ID, JsonObject.mapFrom(ta));
+
+    for (IndividualResource item : items) {
+      var updatedItem = itemsClient.getById(item.getId());
+      assertThat(updatedItem.getJson()
+        .getString("effectiveShelvingOrder"), notNullValue());
+    }
+  }
+
+  @Test
+  public void shouldStopProcessingWhenCannotConvertItem() throws Exception {
+    var item = createItem(0);
+    executeSql("UPDATE item SET jsonb = '{\"a\":\"b\"}'::jsonb WHERE id = '" + item.getId() + "'");
+
+    var ta = getTenantAttributes();
+
+    assertThatThrownBy(() -> tenantOp(TENANT_ID, ta))
+      .hasMessageContaining("Unrecognized field \"a\" " +
+        "(class org.folio.rest.jaxrs.model.Item), not marked as ignorable");
+  }
+
+  private JsonObject getTenantAttributes() {
+    return JsonObject.mapFrom(new TenantAttributes()
+      .withModuleFrom("20.1.1")
+      .withParameters(List.of(
+        new Parameter().withKey("loadSample").withValue("false"),
+        new Parameter().withKey("loadReference").withValue("false"))));
+  }
+
+  private void removeShelvingOrder(List<IndividualResource> items) throws Exception {
+    for (IndividualResource item : items) {
+      unsetJsonbProperty("item", item.getId(), "effectiveShelvingOrder");
+    }
+  }
+
+  private IndividualResource createItem(int index) {
+    var random = new Random();
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    JsonObject itemToCreate = new JsonObject()
+      .put("id", UUID.randomUUID().toString())
+      .put("status", new JsonObject().put("name", "Available"))
+      .put("holdingsRecordId", holdingsRecordId.toString())
+      .put("barcode", random.nextLong())
```

Comment:
There is a small probability that this produces a duplicate barcode causing a failure.
Consider using `UUID.randomUUID().toString()` or `index` or `100000` + `index`.

Reply:
Changed to `100000` + `index`